src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -72,6 +72,7 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
         const resetState = () => dispatch({ type: WalletActions.RESET_STATE })
         provider?.on?.('accountsChanged', resetState)
         provider?.on?.('chainChanged', resetState)
+        const isLocked = !(await provider?._metamask.isUnlocked())
```

Comment:
Don't interact with the MetaMask provider directly but use the hdwallet method mentioned in https://github.com/shapeshift/web/pull/1667#issuecomment-1117228360:
```suggestion
        const isLocked = await wallet.isLocked()
```